src/shared/format-table.h

```diff
@@ -102,9 +102,9 @@ void table_set_cell_height_max(Table *t, size_t height);
 int table_set_empty_string(Table *t, const char *empty);
 int table_set_display_all(Table *t);
 int table_set_display_internal(Table *t, size_t first_column, ...);
-#define table_set_display(...) table_set_display_internal(__VA_ARGS__, (size_t) SIZE_MAX)
+#define table_set_display(...) table_set_display_internal(__VA_ARGS__, SIZE_MAX)
```

Comment:
i think this should stay. this is a varargs function, and we need to make sure that a size_t is put on the stack here. if we don't cast here, it might happen that something smaller is passed here. Now granted, SIZE_MAX is not going to fit in anything smaller, but I think it would still make sense to hardcode the type here, who knows what kind of weird archs there are, where not specifying the type might cause weird stuff to happen.

Reply:
But what exactly? To include the max value, the contant must already be wide enough. If the compiler casts the value to a wider type, no issue. The compiler cannot cast the value to narrower type. And even if it did, it could cast a value with an explicit cast exactly the same. So there is really no scenario where this would make a difference.

Reply:
It should be redundant. I really hope so. But I explicitly added that for safety in my previous PR.

Reply:
Right, I know it was added on purpose. But I don't think we should keep safety measures unless we can show a scenario where they make a difference. As mentioned above, *even* if there was some weird architecture which wanted to pass types as wider than `size_t`, this cast still wouldn't matter.

Reply:
so, on 32bit. SIZE_MAX is going to be 2^31-1. It' not going to fit into "int", hence C would typically pick "long int", i.e. 64bit and put that on the stack. We'd try to read it as size_t though, which is 32bit. On LE that's fine all will work. But on BE this would explode, we'll read the wrong part of it.

i.e. the issue is that SIZE_MAX only fits in *unsigned* native width integers, but I am pretty sure for integers of unspecified signedness C always defaults to *signed* types, and thus might pick one integer size larger than we want.

Reply:
Hmm,  glibc defines SIZE_MAX like this:

```c
/* Limit of `size_t' type.  */
# if __WORDSIZE == 64
#  define SIZE_MAX              (18446744073709551615UL)
# else
#  if __WORDSIZE32_SIZE_ULONG
#   define SIZE_MAX             (4294967295UL)
#  else
#   define SIZE_MAX             (4294967295U)
#  endif
# endif
```

(this is from stdint.h)

i.e. they specify the "U" suffix. i wonder if C99 mandates that though

Reply:
Seems LLVM fucked up they type for SIZE_MAX once upon a time:

https://reviews.llvm.org/D31856

Reply:
So c99 spec says this:

> The  following  object-like macros specify  the  minimum  and  maximum  limits  of integer types corresponding to types defined in other standard headers. Each instance of these macros shall be replaced by a constant expression suitable for use in #if preprocessing directives, and this expression shall have the same type as would an expression that is an object of the corresponding type converted according to the integer promotions.  Its implementation-defined  value  shall  be  equal  to  or  greater  in  magnitude (absolute  value)  than  the  corresponding  value  given below,  with  the  same  sign.

So to me this suggests we can actually rely on SIZE_MAX being of type size_t. So i guess @keszybz changes is OK by C99 standard (even though LLVM once upon a time apparently fucked that up

Reply:
> so, on 32bit. SIZE_MAX is going to be 2^31-1. It' not going to fit into "int", hence C would typically pick "long int", i.e. 64bit and put that on the stack. We'd try to read it as size_t though, which is 32bit. On LE that's fine all will work. But on BE this would explode, we'll read the wrong part of it.

2^3**2**-1. You are right, this would be bad. But in this particular case, it *still* wouldn't matter, because the bit pattern is 0b1111111111…1111111111. So even if read the wrong octets on big-endian, the result would be the same ;)

Thanks for looking into this and merging the patch. If it breaks, the fault is mine ;)